examples/online_serving/whisper_api_client.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: Apache-2.0
+import argparse
+import asyncio
+import json
+from subprocess import CalledProcessError, run
+
+import aiohttp
+import numpy as np
+
+AIOHTTP_TIMEOUT = aiohttp.ClientTimeout(total=6 * 60 * 60)
+SAMPLE_RATE = 16000
+
+
+def load_audio_from_file(file: str, sample_rate: int = SAMPLE_RATE):
+    cmd = [
+        "ffmpeg", "-nostdin", "-threads", "0", "-i", file, "-f", "s16le",
+        "-ac", "1", "-acodec", "pcm_s16le", "-ar",
+        str(sample_rate), "-"
+    ]
+    # fmt: on
+    try:
+        out = run(cmd, capture_output=True, check=True).stdout
+    except CalledProcessError as e:
+        raise RuntimeError(f"Failed to load audio: {e.stderr.decode()}") from e
+
+    return np.frombuffer(out, np.int16).flatten().astype(np.float32) / 32768.0
+
+
+async def iterate_response(response):
+    output_text = ""
+    if response.status == 200:
+        async for chunk_bytes in response.content:
+            chunk_bytes = chunk_bytes.strip()
+            if not chunk_bytes:
+                continue
+            chunk = chunk_bytes.decode("utf-8").removeprefix("data: ")
+            if chunk != "[DONE]":
+                output_text += json.loads(chunk)["text"]
+    return output_text
+
+
+async def transcribe_from_waveform(base_url: str, file_path: str):
```

Comment:
@NickLucche - this matches the API exposed by TritonServer
- https://github.com/triton-inference-server/tensorrtllm_backend/blob/main/tools/whisper/client.py#L144